### PR TITLE
Update galaxy-tool-util to 20.9.0

### PR DESCRIPTION
**Pull request by BiocondaBot**
Update [`galaxy-tool-util`](https://bioconda.github.io/recipes/galaxy-tool-util/README.html): **20.5.0** &rarr; **20.9.0**

[install with bioconda](http://bioconda.github.io/recipes/galaxy-tool-util/README.html) [Conda](https://anaconda.org/bioconda/galaxy-tool-util/files)

Info | Link
-----|-----
Recipe | [`recipes/galaxy-tool-util`](https://github.com//bioconda/bioconda-recipes/tree/bump/galaxy_tool_util/recipes/galaxy-tool-util) (click to view/edit other files)
Releases |[https://pypi.org/pypi/galaxy-tool-util/json](https://pypi.org/pypi/galaxy-tool-util/json)
Recipe Maintainer(s) | @mbargull, @bgruening

***

This pull request was automatically generated (see [docs](https://bioconda.github.io/contributor/updating.html)).

### CODE REVIEW

**Comment by dpryan79** [comment]
Needed for https://github.com/bioconda/bioconda-recipes/pull/23718

**Comment by dpryan79** [comment]
@bgruening @nsoranzo Please have a look at this. There's a file missing from the tarball, so you can't just `pip install` it. I removed the version restriction on galaxy-containers, since it seems that was added to help out the solver rather than as a real dependency (there doesn't seem to be an update to galaxy-containers anyway). It's unclear if that's the correct tack to take or whether there will be an new galaxy-containers release.

File: recipes/galaxy-tool-util/meta.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ build:
     - mulled-build-files=galaxy.tool_util.deps.mulled.mulled_build_files:main
     - mulled-list=galaxy.tool_util.deps.mulled.mulled_list:main
     - mulled-update-singularity-containers=galaxy.tool_util.deps.mulled.mulled_update_singularity_containers:main
-  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+  script: "touch test-requirements.txt && {{ PYTHON }} -m pip install . --no-deps -vv"
```
Comment (nsoranzo) [inline recipes/galaxy-tool-util/meta.yaml:25]:
```suggestion
  # The missing test-requirements.txt will be fixed in https://github.com/galaxyproject/galaxy/pull/10447
  script: "touch test-requirements.txt && {{ PYTHON }} -m pip install . --no-deps -vv"
```

**Comment by nsoranzo** [comment]
Thanks @dpryan79 , I've opened a [PR upstream](https://github.com/galaxyproject/galaxy/pull/10447) to fix this issue, I've added a note here so we remember to drop the workaround when a new version of the package is released.

**Comment by dpryan79** [comment]
Thanks @nsoranzo, anything else obvious that needs to be done or can I merge this after the CI passes?

**Comment by mbargull** [comment]
Thanks for the updates.
> I removed the version restriction on galaxy-containers, since it seems that was added to help out the solver rather than as a real dependency (there doesn't seem to be an update to galaxy-containers anyway). It's unclear if that's the correct tack to take or whether there will be an new galaxy-containers release.

The comment `# Depend on galaxy-containers due to shortcomings of bioconda-utils' build graph generation.` was meant in regards to why this `galaxy-containers` is in `requirements: host:` is at all. It's not needed for the actual build (hence wouldn't need to be in `host` requirements) but IIRC, the build graph only considered build and not run requirements, which is why I added it.
The version restriction on the other hand was added due to discussion at https://github.com/bioconda/bioconda-recipes/pull/19492#discussion_r362050677, i.e., because of the `galaxy-lib`->`galaxy-*` change and it wasn't clear what minimum versions were necessary or if the intend was to have the versions always synced up.
Since `galaxy-containers` did not receive an update, I'd just add the previous version as a lower bound.

File: recipes/galaxy-tool-util/meta.yaml [hunk 2]
```diff
@@ -31,9 +32,9 @@ requirements:
     # Depend on galaxy-util during build time to avoid packaging galaxy/__init__.py .
     - galaxy-util >={{ galaxy_version }}
     # Depend on galaxy-containers due to shortcomings of bioconda-utils' build graph generation.
-    - galaxy-containers >={{ galaxy_version }}
+    - galaxy-containers
   run:
-    - galaxy-containers >={{ galaxy_version }}
+    - galaxy-containers
```
Comment (mbargull) [inline recipes/galaxy-tool-util/meta.yaml:37]:
```suggestion
    - galaxy-containers >=20.5.0
```

**Review by dpryan79 — APPROVED** [review]

**Comment by dpryan79** [comment]
@bioconda-bot please merge

@mbargull Ah, thanks for the context there :)

**Pull request merged**